File: workspace/JavaSelenium2/src/tutorialselenium/PageClasses/FrameworkTestCase.java
package tutorialselenium.PageClasses;

import java.util.concurrent.TimeUnit;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.NoSuchElementException;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.FluentWait;
import org.openqa.selenium.support.ui.Wait;
import org.openqa.selenium.support.ui.WebDriverWait;

import com.google.common.base.Function;

public class FrameworkTestCase {
	private WebDriver driver;
	private String baseUrl;
	SearchPageFactory searchPage;

	@Before
	public void beforeClass() {
		driver = new FirefoxDriver();
		baseUrl = "https://www.expedia.com/";
		
		searchPage = new SearchPageFactory(driver);

		// Maximize the browser's window
		driver.manage().window().maximize();
		driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
		driver.get(baseUrl);
	}

	@Test
	public void test() throws Exception {
		searchPage.clickFlightsTab();
		searchPage.setOriginCity("New York");
		searchPage.setDestinationCity("San Francisco");
		searchPage.setDepartureDate("10/28/2017");
		searchPage.setReturnDate("10/31/2017");
		searchPage.clickSearchButton();
		
	}

	@After
	public void afterClass() throws Exception {
		WebDriverWait wait= new WebDriverWait(driver,60);
		wait.until(ExpectedConditions.invisibilityOfElementLocated(By.id("acol-interstitial")));
		driver.quit();
		Wait<WebDriver> wait1= new FluentWait<WebDriver>(driver).withTimeout(60, TimeUnit.SECONDS).pollingEvery(2, TimeUnit.SECONDS).ignoring(NoSuchElementException.class);
		WebElement display = wait1.until(new Function<WebDriver, WebElement>() {
			public WebElement apply(WebDriver d) {
				return d.findElement(By.id("display"));
			}
		});
	}
}

File: workspace/PythonSelenium/tutorialselenium/FrameworkTestCase.py
'''
Created on 06-Mar-2017

@author: Arjun BR
'''
import unittest
from selenium import webdriver
import time
import SearchPageFactory as searchPage
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.common.by import By

class Test(unittest.TestCase):


    def setUp(self):
        driver= webdriver.Firefox()
        baseUrl = "https://www.expedia.com/";
        driver.maximize_window()
        driver.implicitly_wait(2)
        self.driver=driver
        self.driver.get(baseUrl)

    def testName(self):
        searchPage.clickFlightsTab();
        searchPage.setOriginCity("New York");
        searchPage.setDestinationCity("San Francisco");
        searchPage.setDepartureDate("10/28/2017");
        searchPage.setReturnDate("10/31/2017");
        searchPage.clickSearchButton();
        
        
        
    def tearDown(self):
        WebDriverWait(60).until(EC.invisibility_of_element_located((By.ID, 'acol-interstitial')))
        # WebDriverWait(driver, 10, poll_frequency=1, ignored_exceptions=[ElementNotVisibleException, ElementNotSelectableException])
        time.sleep(2)
        self.driver.quit()
       


    
if __name__ == "__main__":
    #import sys;sys.argv = ['', 'Test.testName']
    unittest.main()